### PR TITLE
Add houston-cleanup-task-usage-data CronJob

### PR DESCRIPTION
**Pull request by rishkarajgi**
## Description

This is a cronjob to hard delete task usage data in houston db older than 90 days

## Related Issues

https://github.com/astronomer/issues/issues/5070

## Testing

Do not merge this PR until this text is replaced with details about how these changes were tested.

## Merging

0.31.0

### CODE REVIEW

File: charts/astronomer/templates/houston/cronjobs/houston-cleanup-task-data-cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+##########################################
+## Houston Cleanup Task Usage data CronJob
+##########################################
+{{- if .Values.houston.cleanupTaskUsageData.enabled }}
+apiVersion: {{ include "apiVersion.batch.cronjob" . }}
+kind: CronJob
+metadata:
+  name: houston-cleanup-task-usage-data
```
Comment (pgvishnuram) [inline charts/astronomer/templates/houston/cronjobs/houston-cleanup-task-data-cronjob.yaml:8]: can be pre-fixed with release name

Comment (danielhoherd) [reply charts/astronomer/templates/houston/cronjobs/houston-cleanup-task-data-cronjob.yaml:8]:
```suggestion
  name: "{{ .Release.Name }}-houston-cleanup-task-usage-data"
```

File: charts/astronomer/templates/houston/cronjobs/houston-cleanup-task-data-cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+{{ toYaml (default .Values.global.platformNodePool.affinity .Values.affinity) | indent 12 }}
+          tolerations:
+{{ toYaml (default .Values.global.platformNodePool.tolerations .Values.tolerations) | indent 12 }}
+          restartPolicy: Never
+{{- include "astronomer.imagePullSecrets" . | indent 10 }}
+          containers:
+            - name: cleanup
+              image: {{ template "houston.image" . }}
+              imagePullPolicy: {{ .Values.images.houston.pullPolicy }}
+              # If you supply only args for a Container, the default Entrypoint defined in the Docker image is run with the args that you supplied.
```
Comment (danielhoherd) [inline charts/astronomer/templates/houston/cronjobs/houston-cleanup-task-data-cronjob.yaml:47]:
```suggestion
```

File: charts/astronomer/templates/houston/cronjobs/houston-cleanup-task-data-cronjob.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  name: "{{ .Release.Name }}-houston-cleanup-task-usage-data"
+  labels:
+    tier: astronomer
+    component: houston-cleanup
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  schedule: {{ .Values.houston.cleanupTaskUsageData.schedule }}
+  # The cron job does not allow concurrent runs; if it is time for a new job run and the previous job run hasn't finished yet, the cron job skips the new job run
```
Comment (danielhoherd) [inline charts/astronomer/templates/houston/cronjobs/houston-cleanup-task-data-cronjob.yaml:17]:
```suggestion
```

**Review by danielhoherd — CHANGES_REQUESTED** [review]
Looks good so far. Needs unit tests.

**Comment by rishkarajgi** [comment]
> Looks good so far. Needs unit tests.

@danielhoherd tests added

cc: @pgvishnuram

**Review by pgvishnuram — COMMENTED** [review]
unit test cases added to verify the flag

**Review by pgvishnuram — APPROVED** [review]

**Pull request merged**